src/test/java/io/jenkins/plugins/coverage/model/CoveragePluginSkipPublishingOfChecksITest.java

```diff
@@ -0,0 +1,183 @@
+
+import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import hudson.model.FreeStyleProject;
+import hudson.model.Run;
+import hudson.plugins.git.GitSCM;
+
+import io.jenkins.plugins.coverage.CoveragePublisher;
+import io.jenkins.plugins.coverage.adapter.JacocoReportAdapter;
+import io.jenkins.plugins.util.IntegrationTestWithJenkinsPerSuite;
+
+import static org.assertj.core.api.Assertions.*;
+
+/**
+ * Tests if publishing of checks can be skipped.
+ */
+public class CoveragePluginSkipPublishingOfChecksITest extends IntegrationTestWithJenkinsPerSuite {
+    private static final String JACOCO_FILENAME = "jacoco-analysis-model.xml";
+    private static final String REPOSITORY_URL = "https://github.com/jenkinsci/analysis-model.git";
+    private static final String COMMIT = "6bd346bbcc9779467ce657b2618ab11e38e28c2c";
+
+    /**
+     * Tests publishing of checks in freestyle-project when skip publishing checks is false.
+     */
+    @Test
+    public void freeStylePublishingOfChecks() {
+        FreeStyleProject project = getFreeStyleProjectWithJacoco(false);
+        checkConsoleLog(buildSuccessfully(project), false);
+    }
+
+    /**
+     * Tests publishing of checks in pipeline-project when skip publishing checks is false.
+     */
+    @Test
+    public void pipelinePublishingOfChecks() {
+        WorkflowJob job = getPipelineProjectWithJacoco(false);
+        checkConsoleLog(buildSuccessfully(job), false);
+    }
+
+    /**
+     * Tests publishing of checks in freestyle-project is skipped when skip publishing checks is true.
+     */
+    @Test
+    public void freeStyleSkipPublishingOfChecks() {
+        FreeStyleProject project = getFreeStyleProjectWithJacoco(true);
+        checkConsoleLog(buildSuccessfully(project), true);
+
+    }
+
+    /**
+     * Tests publishing of checks in pipeline-project is skipped when skip publishing checks is true.
+     */
+    @Test
+    public void pipelineSkipPublishingOfChecks() {
+        WorkflowJob job = getPipelineProjectWithJacoco(true);
+        checkConsoleLog(buildSuccessfully(job), true);
+    }
+
+    /**
+     * Tests publishing of checks with source code when skip publishing checks is false.
+     */
+    @Test
+    public void freeStylePublishingOfChecksWithRepo() throws IOException {
+        FreeStyleProject project = getFreeStyleProjectWithJacoco(false);
+        project.setScm(new GitSCM(REPOSITORY_URL));
+        checkConsoleLog(buildSuccessfully(project), false);
+    }
+
+    /**
+     * Tests publishing of checks with source code when skip publishing checks is false.
+     */
+    @Test
+    public void pipelinePublishingOfChecksWithRepo() {
+        checkConsoleLog(getPipelineProjectWithSCM(false), false);
+    }
+
+    /**
+     * Tests publishing of checks with source code when skip publishing checks is true.
+     */
+    @Test
+    public void freeStyleSkipPublishingOfChecksWithRepo() throws IOException {
```

Comment:
This test will erase your coverage file and will skip the checks because there is no result